myTinyURL/convert/convert.py

```diff
@@ -55,9 +64,14 @@ def short2long(cls, short_id):
             The corresponding long url
             type: str
         """
-        url = get_object_or_404(URL, pk=short_id)
-        url.count += 1
-        url.save()
+        with transaction.atomic():
+            try:
+                url = URL.objects.select_for_update().get(short_id=short_id)
+            except URL.DoesNotExist:
+                raise Http404("Short_id does not exist")
+            # url = get_object_or_404(URL, pk=short_id)
+            url.count += 1
+            url.save()
```

Comment:
You can use a flag/parameter to control whether you want to have synchronization for `short2long()` function. So once you have loadtest framework done, you can easily compare the performance.